extension/lib/cart/ShopgateCartRepository.js

```diff
@@ -0,0 +1,33 @@
+const ShopgateCartBuilder = require('./ShopgateCartBuilder')
+
+class ShopgateCartRepository {
```

Comment:
We can look at this class as an aggregate for a cart, meaning it can only know about single cart and what belongs to it. This knowledge can enable us to simplify implementation - no need for cartId parameter in every method, but instead we'd need an interface with which we can retrieve previously persisted cart identifier.

Reply:
Modified the ShopgateCartRepository so a ShopgateStorage object is passed to it. With this the cart repository is able to look for the cart id using the ShopgateStorage object.